### DISCUSSION TITLE
instalacion de configuracion nvim

### DISCUSSION THREAD

**Discussion by Guaynora** [adalessa/dotfiles#4] [Q&A]
Hola. 
Estoy tratando de tener tu configuración, pero estoy un poco atascado con lo de dap el cual muestra lo siguiente.
```
Error detected while processing /home/jguaynora/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /home/jguaynora/.config/nvim/lua/alpha/dap.lua:1: module 'dap-install' not found:
        no field package.preload['dap-install']
        no file './dap-install.lua'
        no file '/usr/share/luajit-2.1.0-beta3/dap-install.lua'
        no file '/usr/local/share/lua/5.1/dap-install.lua'
        no file '/usr/local/share/lua/5.1/dap-install/init.lua'
        no file '/usr/share/lua/5.1/dap-install.lua'
        no file '/usr/share/lua/5.1/dap-install/init.lua'
        no file './dap-install.so'
        no file '/usr/local/lib/lua/5.1/dap-install.so'
        no file '/usr/lib64/lua/5.1/dap-install.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
        [C]: in function 'require'
        /home/jguaynora/.config/nvim/lua/alpha/dap.lua:1: in main chunk
        [C]: in function 'require'
        /home/jguaynora/.config/nvim/init.lua:19: in main chunk
```
Investigando sobre esto en el repo del paquete veo que hay una nueva versión, pero que no esta como terminada y no hay una documentación clara y algunos mencionan que es mejor usar la rama dev hasta que la main este completa.
Pero veo que en la documentación de la rama dev se menciona la instalación con Plug y estoy un poco confundido con respecto a lua para realizar la instalación de este dap

**Comment by Guaynora** [comment]
> Hola. Estoy tratando de tener tu configuración, pero estoy un poco atascado con lo de dap el cual muestra lo siguiente.
> 
> ```
> Error detected while processing /home/jguaynora/.config/nvim/init.lua:
> E5113: Error while calling lua chunk: /home/jguaynora/.config/nvim/lua/alpha/dap.lua:1: module 'dap-install' not found:
>         no field package.preload['dap-install']
>         no file './dap-install.lua'
>         no file '/usr/share/luajit-2.1.0-beta3/dap-install.lua'
>         no file '/usr/local/share/lua/5.1/dap-install.lua'
>         no file '/usr/local/share/lua/5.1/dap-install/init.lua'
>         no file '/usr/share/lua/5.1/dap-install.lua'
>         no file '/usr/share/lua/5.1/dap-install/init.lua'
>         no file './dap-install.so'
>         no file '/usr/local/lib/lua/5.1/dap-install.so'
>         no file '/usr/lib64/lua/5.1/dap-install.so'
>         no file '/usr/local/lib/lua/5.1/loadall.so'
> stack traceback:
>         [C]: in function 'require'
>         /home/jguaynora/.config/nvim/lua/alpha/dap.lua:1: in main chunk
>         [C]: in function 'require'
>         /home/jguaynora/.config/nvim/init.lua:19: in main chunk
> ```
> 
> Investigando sobre esto en el repo del paquete veo que hay una nueva versión, pero que no esta como terminada y no hay una documentación clara y algunos mencionan que es mejor usar la rama dev hasta que la main este completa. Pero veo que en la documentación de la rama dev se menciona la instalación con Plug y estoy un poco confundido con respecto a lua para realizar la instalación de este dap


Pude resolver este incoviente yendo al archivo dentro de la carpeta packer y cambiando la rama.
Pero me encuentro con otros problemas que me aparecen sobre null-ls.lua y reesitter.lua y no sé si se deba a php, yo no programo con php y no sé si es necesario que tenga que instalar php para poder solventar este problema.
```
Error detected while processing /home/jguaynora/.dotfiles/nvim/.config/nvim/after/plugin/null-ls.lua:
E5113: Error while calling lua chunk: /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:25: no parser for 'phpdoc' language, see :help treesitter-parsers
stack traceback:
        [C]: in function 'error'
        /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:25: in function 'require_language'
        /usr/share/nvim/runtime/lua/vim/treesitter/query.lua:166: in function 'parse_query'
        /usr/share/nvim/runtime/lua/vim/treesitter/query.lua:119: in function 'set_query'
        /home/jguaynora/.config/nvim/lua/alpha/php_code_actions.lua:25: in main chunk
        [C]: in function 'require'
        ...ora/.dotfiles/nvim/.config/nvim/after/plugin/null-ls.lua:14: in main chunk
Error detected while processing /home/jguaynora/.dotfiles/nvim/.config/nvim/after/plugin/treesitter.lua:
tree-sitter CLI not found: `tree-sitter` is not executable!

```

**Reply by adalessa** [reply]
Perdon por la demora en la respuesta, github no me aviso de que se inicio una discucion. 
Con respecto a lo de phpdoc si es por eso, fijate aca https://github.com/adalessa/dotfiles/blob/main/nvim/.config/nvim/after/plugin/null-ls.lua#L11 podes comentar o borrar directamente.
Lo de tree-sitter no recuerdo haber tenido ese problema. Todavia te sucede.

De nuevo disculpa por la demora. Saludos

**Reply by Guaynora** [reply]
No se preocupe, al final termine haciendo una configuración desde cero y guiandome un poco de sus configuraciones.
Saludos
